### PR TITLE
Feature/pn-38

### PR DESCRIPTION
**Pull request by tankakatan**

### CODE REVIEW

**Comment by tankakatan** [comment]
@brianunlam looks like the packaged executable is broken. When I run it I get the following error:
```denis@MacBook-Bro:~$ DATADIR=~/.point ./Downloads/pointnetwork-macos-2
WARNING: No configurations found in configuration directory:/Users/denis/config
WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.
pkg/prelude/bootstrap.js:1740
      throw error;
      ^

Error: Configuration property "datadir" is not defined
    at Config.get (/snapshot/pointnetwork/node_modules/config/lib/config.js:182:11)
    at Object.<anonymous> (/snapshot/pointnetwork/dist/core/log.js:10:24)
    at Module._compile (pkg/prelude/bootstrap.js:1794:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at Module.require (pkg/prelude/bootstrap.js:1719:31)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/snapshot/pointnetwork/dist/client/zweb/deployer/index.js:6:16)
```
Probably I broke something.

File: src/@types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        deploy?: boolean,
+        deploy_contracts?: boolean,
+        dev?: boolean,
+    };
+
+    type CtxType = Record<string, unknown> & {
+        basepath: string;
+        exit: (code: number) => void;
+        die: (err: Error) => void;
+        db?: {shutdown?: () => Promise<undefined>};
```
Comment (brianunlam) [inline src/@types/index.d.ts:36]: I think void is better than undefined for this case.

```
Promise<void>
```

File: src/db/models/index.ts [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+
 const log = logger.child({module: 'Sequelize'});
-const {resolveHome} = require('../../core/utils');
 
-class SequelizeFactory {
-    init(ctx) {
-        this.ctx = ctx;
+interface SequelizeFactory {
+    config: DatabaseConfig;
+    Sequelize: typeof Sequelize;
```
Comment (brianunlam) [inline src/db/models/index.ts:13]: maybe we should install types for Sequelize

**Review by brianunlam — APPROVED** [review]
LGTM

File: src/db/models/index.ts (hunk 2 above)
Comment (tankakatan) [reply src/db/models/index.ts:13]: Already did [here](https://github.com/pointnetwork/pointnetwork/pull/236/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R76).

**Pull request merged**